torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -170,6 +173,8 @@
 GEMM_TEMPLATE_STUB_DEF = r"""
 {%- if x_scale is not none %}
     {%- set kernel_args = {"X": X, "W": W, "inp": inp, "x_scale": x_scale, "x_zp": x_zp, "w_scale": w_scale, "w_zp": w_zp,} %}
+{%- elif is_woq_int4 %}
+    {%- set kernel_args = {"X": X, "W": W, "q_group_size": q_group_size, "qscale_and_zeros": qscale_and_zeros} %}
```

Comment:
Did you explore caching scales & zeros separately? Thanks

cc @leslie-fang-intel

Reply:
What do you mean by caching scales & zeros separately?

Reply:
Caching them in separate tensors

Reply:
What is the benefit of do that?

Reply:
This computation would be avoided:

<img width="545" alt="image" src="https://github.com/user-attachments/assets/ea93f30a-9e63-4b8a-8fa2-63257d6f2220" />

Reply:
Not sure if the performance difference would be discernible, though. Probably not.